internal/services/pam/pam_test.go

```diff
@@ -0,0 +1,606 @@
+package pam_test
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd"
+	"github.com/ubuntu/authd/cmd/authd/daemon"
+	"github.com/ubuntu/authd/internal/testutils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var configFmt = `
+systemdirs:
+    socketpath: %s
+    brokercfgdir: %s
+`
+
+var (
+	fixturesPath = filepath.Join("testdata", "fixtures")
+)
+
+func TestAvailableBrokers(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		cfgDir string
+
+		wantErr bool
+	}{
+		"All brokers when config dir only has valid brokers":      {cfgDir: "valid_brokers"},
+		"Only correct brokers when config dir has mixed brokers":  {cfgDir: "mixed_brokers"},
```

Comment:
what is a mixed broker config directory? more seriously, I would be a little bit descriptive, some valid, some invalid or so…

Reply:
Yeah... The "very" descriptive name won't work as we get over the limit for the file path in the socket :smiling_face_with_tear: